test/k8sT/Services.go

```diff
@@ -412,6 +412,144 @@ var _ = Describe("K8sServicesTest", func() {
 		})
 	})
 
+	SkipContextIf(func() bool { return !helpers.RunsOnNetNextOr419Kernel() }, "Checks local redirect policy", func() {
+		const (
+			lrpServiceName = "lrp-demo-service"
+			fe1Name        = "k8s1-client"
+			fe2Name        = "k8s2-client"
+			be1Name        = "k8s1-backend"
+			be2Name        = "k8s2-backend"
+			repeat         = 10
+			feFilter       = "role=frontend"
+			beFilter       = "role=backend"
+		)
+
+		var (
+			deploymentYAML string
+			lrpSvcYAML     string
+			ctx            context.Context
+			ctxCancel      context.CancelFunc
+			svcIP          string
+			curlTCP        string
+			curlUDP        string
+		)
+
+		BeforeAll(func() {
+			deploymentYAML = helpers.ManifestGet(kubectl.BasePath(), "lrp-test.yaml")
+			lrpSvcYAML = helpers.ManifestGet(kubectl.BasePath(), "lrp-svc.yaml")
+			res := kubectl.ApplyDefault(deploymentYAML)
+			res.ExpectSuccess("Unable to apply %s", deploymentYAML)
+			pods := []string{feFilter, beFilter}
+			for _, pod := range pods {
+				err := kubectl.WaitforPods(helpers.DefaultNamespace, fmt.Sprintf("-l %s", pod), helpers.HelperTimeout)
+				ExpectWithOffset(1, err).Should(BeNil())
+			}
+			ctx, ctxCancel = context.WithTimeout(context.Background(), 1*time.Second)
+			clusterIP, _, err := kubectl.GetServiceHostPort(helpers.DefaultNamespace, lrpServiceName)
+			svcIP = clusterIP
+			Expect(err).To(BeNil(), "Cannot get svc IP")
+			httpSVCURL := fmt.Sprintf("http://%s/", svcIP)
+			tftpSVCURL := fmt.Sprintf("tftp://%s/hello", svcIP)
+			curlTCP = helpers.CurlFail(httpSVCURL)
+			curlUDP = helpers.CurlFail(tftpSVCURL)
+		})
+
+		AfterAll(func() {
+			_ := kubectl.Delete(deploymentYAML)
+			ctxCancel()
+		})
+
+		It("LRP connectivity", func() {
+			// Basic sanity check
+			ciliumPods, err := kubectl.GetCiliumPods()
+			Expect(err).To(BeNil(), "Cannot get cilium pods")
+			for _, pod := range ciliumPods {
+				service := kubectl.CiliumExecMustSucceed(context.TODO(), pod, fmt.Sprintf("cilium service list | grep %s", svcIP), "Cannot retrieve services on cilium pod")
+				service.ExpectContains("LocalRedirect", "LocalRedirect is not present in the cilium service list")
+			}
+
+			By("Checking traffic goes to local backend")
+			testCases := []struct {
+				selector string
+				pod      string
+				want     string
+				notWant  string
+			}{
+				{
+					selector: "id=app1",
+					pod:      fe1Name,
+					// Expects to see local backend name in returned Hostname field
+					want: be1Name,
+					// Expects never to see remote backend name in returned Hostname field
+					notWant: be2Name,
+				},
+				{
+					selector: "id=app2",
+					pod:      fe2Name,
+					want:     be2Name,
+					notWant:  be1Name,
+				},
+			}
+			for _, tc := range testCases {
+				for try := 0; try < repeat; try++ {
```

Comment:
Why do we need to do retries? Is this something that will eventually succeed? If so, then please use `Eventually` for this.

Reply:
hmm, not sure if `Eventually` is suitable here. This is to check that lb will "always" send traffic to local backend and "never" send it to remote backend.

Reply:
Ah, I see, so then it's the opposite! We can use Consistently to do this :wink:

Reply:
makes sense, I think consistently would work for this test, but I doubt either `consistently` or `eventually` matches the semantics in the restore test (line 533), where we need results from multiple curl results to "sometimes" have substring1 and "sometimes" have substring2, any magic work/pattern to use there?

Reply:
ah is that just  an `Eventually` && `Eventually` case?

Reply:
Done.